common/tasks/task.go

```diff
@@ -53,6 +53,21 @@ type (
 	}
 )
 
+func (left Key) CompareTo(right Key) int {
+	if left.FireTime.Before(right.FireTime) {
```

Comment:
do we need a CompareTo that is more approximate?  Two timers that are one nanosecond apart might just as well be considered equal in many contexts.

Reply:
e.g. 2 timers with same firing time, but targeting at different workflows.

plus building block logic should be as strict as possible